_posts/2018-01-24-Java9-modules-Spring-Boot-2-Docker.md

```diff
@@ -0,0 +1,301 @@
+COPY target/*.jar /opt/
+
+ENV JAVA_HOME=/opt/jdk-9-minimal
+ENV PATH="$PATH:$JAVA_HOME/bin"
+
+EXPOSE 8080
+CMD java -jar /opt/*.jar
+```
+
+Nice, isn't it? There are still leaner technologies to use for micro-services, but why not grab the benefits of Java platform technological advances?
+
+You can get a full listing of modules by running:
+
+```bash
+➜  my-crazy-machine git:(master) ✗ java --list-modules
+java.activation@9
+java.base@9
+java.compiler@9
+java.corba@9
+java.datatransfer@9
+java.desktop@9
+java.instrument@9
+java.jnlp@9
+java.logging@9
+java.management@9
+java.management.rmi@9
+java.naming@9
+java.prefs@9
+java.rmi@9
+java.scripting@9
+java.se@9
+java.se.ee@9
+java.security.jgss@9
+java.security.sasl@9
+java.smartcardio@9
+java.sql@9
+java.sql.rowset@9
+java.transaction@9
+java.xml@9
+java.xml.bind@9
+java.xml.crypto@9
+java.xml.ws@9
+java.xml.ws.annotation@9
+javafx.base@9
+javafx.controls@9
+javafx.deploy@9
+javafx.fxml@9
+javafx.graphics@9
+javafx.media@9
+javafx.swing@9
+javafx.web@9
+jdk.accessibility@9
+jdk.attach@9
+jdk.charsets@9
+jdk.compiler@9
+jdk.crypto.cryptoki@9
+jdk.crypto.ec@9
+jdk.deploy@9
+jdk.deploy.controlpanel@9
+jdk.dynalink@9
+jdk.editpad@9
+jdk.hotspot.agent@9
+jdk.httpserver@9
+jdk.incubator.httpclient@9
+jdk.internal.ed@9
+jdk.internal.jvmstat@9
+jdk.internal.le@9
+jdk.internal.opt@9
+jdk.internal.vm.ci@9
+jdk.jartool@9
+jdk.javadoc@9
+jdk.javaws@9
+jdk.jcmd@9
+jdk.jconsole@9
+jdk.jdeps@9
+jdk.jdi@9
+jdk.jdwp.agent@9
+jdk.jfr@9
+jdk.jlink@9
+jdk.jshell@9
+jdk.jsobject@9
+jdk.jstatd@9
+jdk.localedata@9
+jdk.management@9
+jdk.management.agent@9
+jdk.management.cmm@9
+jdk.management.jfr@9
+jdk.management.resource@9
+jdk.naming.dns@9
+jdk.naming.rmi@9
+jdk.net@9
+jdk.pack@9
+jdk.packager@9
+jdk.packager.services@9
+jdk.plugin@9
+jdk.plugin.dom@9
+jdk.plugin.server@9
+jdk.policytool@9
+jdk.rmic@9
+jdk.scripting.nashorn@9
+jdk.scripting.nashorn.shell@9
+jdk.sctp@9
+jdk.security.auth@9
+jdk.security.jgss@9
+jdk.snmp@9
+jdk.unsupported@9
+jdk.xml.bind@9
+jdk.xml.dom@9
+jdk.xml.ws@9
+jdk.zipfs@9
+oracle.desktop@9
+oracle.net@9
+```
+
+So with full-blown non-trimmed JDK, that's what you have on the disk, and that's pretty much what you load in memory every time you run a hello-world. In the future we will all be tailoring our Java, pretty much.
+
+Unfortunately as of right now, Spring Boot is not very modular, in sense of Java 9 modules. It's still much jar-based, and comes with a metric drek ton of dependencies. When running Spring, whatever you save in JDK/JRE memory footprint, will have less of an effect on full service. But I expect this to change, in the future, when more people start using later versions of technology. There are also more lightweight runtimes around, with less core features, such as Java Microprofile, and Spark. If you're looking to really put Java on a diet, you might want to take a look at these, too. Both are very interesting and attractive choices.
```

Comment:
Unfortunately, Spring Boot is not particularly modular  in the sense of Java 9 modules. ?